controllers/managedfusionoffering_controller.go

```diff
@@ -71,21 +93,169 @@ func (r *ManagedFusionOfferingReconciler) Reconcile(ctx context.Context, req ctr
 
 func (r *ManagedFusionOfferingReconciler) initReconciler(ctx context.Context, req ctrl.Request) {
 	r.ctx = ctx
+	r.namespace = req.Namespace
 
 	r.managedFusionOffering = &v1alpha1.ManagedFusionOffering{}
 	r.managedFusionOffering.Name = req.Name
 	r.managedFusionOffering.Namespace = req.Namespace
+
+	r.ocsOperatorGroup = &opv1.OperatorGroup{}
+	r.ocsOperatorGroup.Name = operatorGroupName
+	r.ocsOperatorGroup.Namespace = req.Namespace
+
+	r.ocsCatalogSource = &opv1a1.CatalogSource{}
+	r.ocsCatalogSource.Name = catalogSourceName
+	r.ocsCatalogSource.Namespace = req.Namespace
+
+	r.ocsSubscription = &opv1a1.Subscription{}
+	r.ocsSubscription.Name = subscriptionName
+	r.ocsSubscription.Namespace = req.Namespace
+}
+
+func (r *ManagedFusionOfferingReconciler) reconcilePhases() (reconcile.Result, error) {
+	if err := r.reconcileOperatorGroup(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := r.reconcileCatalogSource(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := r.reconcileSubscription(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := r.reconcileInstallPlan(); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
 }
 
 // This function is a placeholder for offering plugin integration
 func pluginReconcile(r *ManagedFusionOfferingReconciler) (ctrl.Result, error) {
 	return ctrl.Result{}, nil
 }
 
+func (r *ManagedFusionOfferingReconciler) reconcileOperatorGroup() error {
+	r.Log.Info("Reconciling DF Offering Operator Group")
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.ocsOperatorGroup, func() error {
+		if err := r.own(r.ocsOperatorGroup); err != nil {
+			return err
+		}
+		r.ocsOperatorGroup.Spec.TargetNamespaces = []string{r.namespace}
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
- if we aren't customising `err` then wouldn't `return err` suffice?

Reply:
updated